webapp/channels/src/components/admin_console/generated_setting.tsx

```diff
@@ -38,7 +38,11 @@ export default class GeneratedSetting extends React.PureComponent<Props> {
     private regenerate = (e: React.MouseEvent) => {
         e.preventDefault();
 
-        this.props.onChange(this.props.id, crypto.randomBytes(256).toString('base64').substring(0, 32));
+        // Pure base64 implementation can contain characters that are not URL safe without additional
+        // encoding. Adopt a URL/Filename safer alphabet as noted in https://datatracker.ietf.org/doc/html/rfc4648#section-5
+        // where: 62 - (minus) , 63 _ (underscore)
+        const value = crypto.randomBytes(256).toString('base64').substring(0, 32);
+        this.props.onChange(this.props.id, value.replaceAll('+', '-').replaceAll('/', '_'));
```

Comment:
We can do this with one call to `replaceAll` using the below regex, though 0/5 on this, and on how the code should be organized here

```suggestion
        let value = crypto.randomBytes(256).toString('base64').substring(0, 32);
        value = value.replaceAll(/[+/]/g, '_');
        this.props.onChange(this.props.id, value);
```

Reply:
The PR's code is a bit different in that it changes to dashes and underscores rather than just underscores. I think would actually prefer changing dashes to underscores as well, as it makes the secret easier to triple-click and copy.

Reply:
I saw some value in using a standard variant of the base64 standard rather than using our own encoding + triple-click is a bit of a bit power-user behavior.

But if you feel that's very important for you, I can follow your path.

Reply:
We can keep it as-is :+1:

Reply:
I would like @DHaussermann's opinion on the double/triple click topic